### PR TITLE
Improve font handling

### PR DESCRIPTION
**Pull request by Xenoamor**
Looking into closing #80 I realised that a lot needed to change. Here are my changes, all tested and appears to be working:

**API changes (changes that coders need to be aware of) :**
1. ~~All units passed to `uui_number` are treated as milli (a factor of one thousand). This means you no longer have to divide numbers to get them to display correctly like was previously done with `input_voltage` and `output_voltage`~~ See later comment about SI prefix
2. ~~`number_draw()` now draws from left to right rather than right to left. This is in keeping with how the icon/graphic drawing works~~ See later comment about alignment
3. `ui_number_t` now has a boolean called `pad_dot`. Setting this to true will make the '.' character the same width as the digits like in the original DPS5005 firmware. This does the following:
image
Please not that this option is set to false on all current screens
4. `gen_lookup.py` now uses `-i` to specify an image not sharing `-f` with fonts. This is how it should have been
5. `gen_lookup.py` now generates a `#define` of the amount of pixels to space characters by when they are drawn. This can be specified with the -sp <pixels> flag, or left to be generated by the tool

**Bug fixes and general changes:**
1. Better character width detection now prevents clipping/overlap between glyphs of varying sizes
2. Removed black_buffer from `tft.c` in favour of `tft_fill(BLACK)`
3. `ili9163c_draw_vline()` and `ili9163c_draw_hline()` have been tested to work and are now in place
4. A bug where some focused character highlighting borders were left when different width characters were used has been fixed. This has been done by handling this functionality in `uui_number.c` not `tft.c`
5. Font highlighting now fills the entire bounding box. To me this makes more sense as the high contrast flickering caused by rapidly changing between different width digits was distracting
6. `gen_lookup.py` now generates a `#define` of the max digit width. This is useful as digits usually vary in width to the capital V and A and it allows for packing them more closely together in `uui_number.c`
7. `gen_lookup.py` now generates a `#define` of the dot width. This is useful for spacing things in `uui_number.c`

### CODE REVIEW

**Comment by Xenoamor** [comment]
I think this could be improved to allow right or left alignment in `uui_number.c`

**Comment by Xenoamor** [comment]
Please don't merge this until I have the right alignment in <3

**Comment by Xenoamor** [comment]
Okay I believe this is all good to be reviewed @kanflo 
You can now right align:
image
And you can also draw numbers with a decimal place:
image

You can also remove the unit character if you so like

File: opendps/func_cc.c [hunk 1]
```diff
@@ -180,7 +186,7 @@ static void cc_enable(bool enabled)
 {
     if (enabled) {
         saved_i = cc_current.value;
-        (void) pwrctl_set_vout(10 * cc_voltage.max - 1000);
+        (void) pwrctl_set_vout(cc_voltage.max - 1000); /** Updated in cc_tick */
```
Comment (kanflo) [inline opendps/func_cc.c]: Hmm, magic number 1000 is probably a leftover now that the max value is calculated on line 259. Should be removed at some point.

File: opendps/gen_lookup.py [hunk 2]
```diff
@@ -123,17 +123,26 @@ def convert_font_to_c(font_fname, characters, character_strings, font_size, outp
...
+        if width > digit_max_width and characters[i].isdigit():
+            digit_max_width = width # Find the maximum digit width
         count += 1
         if count < len(characters):
             font_source_file.write(",\n  ")
     font_source_file.write("\n};\n\n")
 
+    # If no spacing parameter is set then generate one based on the max glyph width
+    if not args.font_spacing:
+        args.font_spacing = int(round(0.223 * float(character_max_width) - 0.863)) # This is fairly rough
```
Comment (kanflo) [inline opendps/gen_lookup.py]: And fairly magic ;)

**Comment by kanflo** [comment]
Looks good! For some reason `GenericMobileSystem.ttf` was drawn too far to the left:

img_4441

but that can be worked out later, I will merge this fine PR.

File: opendps/gen_lookup.py (hunk 2 above)
Comment (Xenoamor) [reply opendps/gen_lookup.py]: I visually spaced a pile of different fonts and drew a graph from it :L

File: opendps/func_cc.c (hunk 1 above)
Comment (Xenoamor) [reply opendps/func_cc.c]: I think the original DPS firmware sets the maximum output voltage as Vin - 1V

**Comment by kanflo** [comment]
BTW. Was the leading zero on the voltage intentional?

File: opendps/gen_lookup.py (hunk 2 above)
Comment (kanflo) [reply opendps/gen_lookup.py]: Wow!

**Comment by Xenoamor** [comment]
> Looks good! For some reason `GenericMobileSystem.ttf` was drawn too far to the left:

It is likely because the font itself has wider characters. You can either move the x position further right or reduce the large font size `FONT_LARGE_SIZE` in the makefile

I thought I had fixed the leading zero issue. I'll have another look into it now

**Comment by Xenoamor** [comment]
#107 should resolve this

**Pull request merged**